sdk/src/test/java/com/finbourne/lusid/integration/AggregationTests.java

```diff
@@ -0,0 +1,224 @@
+                            .transactionId(UUID.randomUUID().toString())
+                            .type("FundsIn")
+                            .instrumentUid("CCY_GBP")
+                            .totalConsideration(new CurrencyAndAmount().currency("GBP").amount(0.0))
+                            .transactionDate(EFFECTIVE_DATE)
+                            .settlementDate(EFFECTIVE_DATE)
+                            .units(30600.0)
+                            .source(TransactionRequest.SourceEnum.CLIENT)
+                    );
+
+                    return requests;
+                },
+                results -> {
+                    assertEquals(3, results.size());
+                    assertEquals(20000.0, results.get(0).get(AGGREGATION_KEY));
+                    assertEquals(30000.0, results.get(1).get(AGGREGATION_KEY));
+                    assertEquals(10000.0, results.get(2).get(AGGREGATION_KEY));
+                });
+    }
+
+    public void Run_Aggregation(
+            Supplier<List<TransactionRequest>> createTransactionRequests,
+            Consumer<List<Map<String, Object>>> validateResults
+    ) throws ApiException
+    {
+        String uuid = UUID.randomUUID().toString();
+        String scope = "finbourne";
+
+        //  portfolio request
+        String originalPortfolioId = String.format("Id-%s", uuid);
+        CreateTransactionPortfolioRequest request = new CreateTransactionPortfolioRequest()
+                .displayName(String.format("Portfolio-%s", uuid))
+                .code(originalPortfolioId)
+                .baseCurrency("GBP")
+                .created(EFFECTIVE_DATE);
+
+        //  create portfolio
+        Portfolio portfolio = this.transactionPortfoliosApi.createPortfolio(scope, request);
+
+        assertEquals(portfolio.getId().getCode(), originalPortfolioId);
+
+        String portfolioId = portfolio.getId().getCode();
+
+        //  add the transactions
+        List<TransactionRequest> transactionRequests = createTransactionRequests.get();
+
+        //  add the trades
+        this.transactionPortfoliosApi.upsertTransactions(scope, portfolioId, transactionRequests);
+
+        //  set up the prices used for the aggregation in the analytic stores
+        ResourceListOfAnalyticStoreKey analyticsStores = this.analyticsStoresApi.listAnalyticStores(null, null, null, null, null);
+        AnalyticStoreKey analyticStore = analyticsStores.getValues()
+                .stream()
+                .filter(ask -> ask.getDate().equals(EFFECTIVE_DATE))
+                .findAny()
+                .orElse(null);
+
+        if (analyticStore == null) {
+            //  create the analytic store
+            CreateAnalyticStoreRequest  createAnalyticStoreRequest = new CreateAnalyticStoreRequest().scope(scope).date(EFFECTIVE_DATE);
+            this.analyticsStoresApi.createAnalyticStore(createAnalyticStoreRequest);
+        }
+
+        List<InstrumentAnalytic>    prices = List.of(
```

Comment:
"What do you do if you see a space-man?"
"Park in it, man"

(Unnecessary spaces)